sqlglot/optimizer/simplify.py

```diff
@@ -1070,3 +1067,69 @@ def _flat_simplify(expression, simplifier, root=True):
                 lambda a, b: expression.__class__(this=a, expression=b), operands
             )
     return expression
+
+
+def gen(expression: t.Any) -> str:
+    """Simple pseudo sql generator for quickly generating sortable and uniq strings.
+
+    Sorting and deduping sql is a necessary step for optimization. Calling the actual
+    generator is expensive so we have a bare minimum sql generator here.
```

Comment:
why is calling the generator expensive? can't we set `copy=False` or something? this seems do add quite some complexity

Reply:
generator is super expensive, so i needed to add a cache

Reply:
we can't bring it back?

Reply:
it’s not necessary anymore and removes complexity